packages/x-data-grid/src/components/cell/GridSkeletonCell.tsx

```diff
@@ -5,45 +5,101 @@ import {
+  string: [40, 80],
+  date: [40, 60],
+  dateTime: [60, 80],
+  singleSelect: [40, 80],
+} as const;
+
+export interface GridSkeletonCellProps extends React.HTMLAttributes<HTMLDivElement> {
+  type?: GridColType;
+  width?: number | string;
+  height?: number | 'auto';
+  field?: string;
+  align?: string;
+  /**
+   * If `true`, the cell will not display the skeleton but still reserve the cell space.
+   * @default false
+   */
+  empty?: boolean;
 }
 
-type OwnerState = Pick<GridSkeletonCellProps, 'align'> & {
+type OwnerState = Pick<GridSkeletonCellProps, 'align' | 'empty'> & {
   classes?: DataGridProcessedProps['classes'];
 };
 
 const useUtilityClasses = (ownerState: OwnerState) => {
-  const { align, classes } = ownerState;
+  const { align, classes, empty } = ownerState;
 
   const slots = {
-    root: ['cell', 'cellSkeleton', `cell--text${capitalize(align)}`, 'withBorderColor'],
+    root: [
+      'cell',
+      'cellSkeleton',
+      `cell--text${align ? capitalize(align) : 'Left'}`,
+      empty && 'cellEmpty',
+    ],
   };
 
   return composeClasses(slots, getDataGridUtilityClass, classes);
 };
 
-function GridSkeletonCell(props: React.HTMLAttributes<HTMLDivElement> & GridSkeletonCellProps) {
-  const { field, align, width, height, ...other } = props;
+const randomNumberGenerator = createRandomNumberGenerator(12345);
+
+function GridSkeletonCell(props: GridSkeletonCellProps) {
+  const { field, type, align, width, height, empty = false, style, className, ...other } = props;
   const rootProps = useGridRootProps();
-  const ownerState = { classes: rootProps.classes, align };
+  const ownerState = { classes: rootProps.classes, align, empty };
   const classes = useUtilityClasses(ownerState);
-  const contentWidth = Math.round(randomWidth());
+
+  // Memo prevents the non-circular skeleton widths changing to random widths on every render
+  const skeletonProps = React.useMemo(() => {
+    const isCircularContent = type === 'boolean' || type === 'actions';
+
+    if (isCircularContent) {
+      return {
+        variant: 'circular',
+        width: CIRCULAR_CONTENT_SIZE,
+        height: CIRCULAR_CONTENT_SIZE,
+      } as const;
+    }
+
+    // The width of the skeleton is a random number between the min and max values
+    // The min and max values are determined by the type of the column
+    const [min, max] = type
+      ? CONTENT_WIDTH_RANGE_BY_TYPE[type] ?? DEFAULT_CONTENT_WIDTH_RANGE
+      : DEFAULT_CONTENT_WIDTH_RANGE;
+
+    return {
+      variant: 'text',
+      width: `${Math.round(randomNumberGenerator(min, max))}%`,
+      height: CONTENT_HEIGHT,
+    } as const;
+  }, [type]);
 
   return (
-    <div className={classes.root} style={{ height, maxWidth: width, minWidth: width }} {...other}>
-      <Skeleton width={`${contentWidth}%`} height={25} />
+    <div
+      data-field={field}
+      className={clsx(classes.root, className)}
+      style={{ height, maxWidth: width, minWidth: width, ...style }}
```

Comment:
**Off-topic**

I never quite understood why Damien first wrote the cells with `minWidth`+ `maxWidth` instead of only `width`. I suspect a simplification opportunity.

Reply:
Oh, interesting, I have tried to reproduce the issue and couldn't, so #14448 😁, I don't know, why not;